src/reducers/ProgressTrackerReducer.jsx

```diff
@@ -0,0 +1,67 @@
+const initialState = {
+  genreqs: 0,
+  majorreqs: 0,
+};
+
+function sum(list) {
+  let total = 0;
+  for (let i = 0, len = list.length; i < len; i += 1) {
+    total += list[i].credits;
+  }
+  return total;
+}
+
+const ProgressTrackerReducer = (state = initialState, action) => {
+  const mathCourses = [];
+  const engrCourses = [];
+  const sciCourses = [];
+  const ahseCourses = [];
+
+  switch (action.type) {
+    case 'GET_COMPLETED_COURSES': {
+      const data = action.data;
+      let genreqs = data.map(a => a.generalRequirements[0]);
+      genreqs = genreqs.filter(n => n !== undefined);
+      let majorreqs = data.map(a => a.otherRequirements[0]);
+      majorreqs = majorreqs.filter(n => n !== undefined);
+
+      for (let i = 0; i < data.length; i += 1) {
+        if (data[i].registrarId.substring(0, 3) === 'MTH') {
```

Comment:
currently, the way the database is, I think we have to do the substring operation. Its only four if statements. A switch statement might be faster but the difference is negligible